warp10/src/main/java/io/warp10/script/functions/TYPEOF.java

```diff
@@ -97,57 +108,72 @@ public static String typeof(Object o) {
     } else if (List.class.isAssignableFrom(c)) {
-      return (TYPE_LIST);
+      return TYPE_LIST;
     } else if (Map.class.isAssignableFrom(c)) {
-      return (TYPE_MAP);
+      return TYPE_MAP;
     } else if (Macro.class.isAssignableFrom(c)) {
-      return (TYPE_MACRO);
-    } else if (WarpScriptMapperFunction.class.isAssignableFrom(c)) {
-      return (TYPE_MAPPER);
-    } else if (WarpScriptReducerFunction.class.isAssignableFrom(c)) {
-      return (TYPE_REDUCER);
+      return TYPE_MACRO;
+    } else if (WarpScriptAggregatorFunction.class.isAssignableFrom(c)) {
+      return TYPE_AGGREGATOR;
+    } else if (WarpScriptFillerFunction.class.isAssignableFrom(c)) {
+      return TYPE_FILLER;
+    } else if (WarpScriptFilterFunction.class.isAssignableFrom(c)) {
+      return TYPE_FILTER;
+    } else if (WarpScriptNAryFunction.class.isAssignableFrom(c)) {
+      return TYPE_OPERATOR;
     } else if (GeoTimeSerie.class.isAssignableFrom(c)) {
-      return (TYPE_GTS);
+      return TYPE_GTS;
+    } else if (GTSEncoder.class.isAssignableFrom(c)) {
+      return TYPE_GTSENCODER;
     } else if (byte[].class.isAssignableFrom(c)) {
-      return (TYPE_BYTES);
+      return TYPE_BYTES;
     } else if (PGraphics.class.isAssignableFrom(c)) {
-      return (TYPE_PGRAPHICSIMAGE);
+      return TYPE_PGRAPHICSIMAGE;
     } else if (PImage.class.isAssignableFrom(c)) {
-      return (TYPE_PIMAGE);
+      return TYPE_PIMAGE;
     } else if (PFont.class.isAssignableFrom(c)) {
-      return (TYPE_PFONT);
+      return TYPE_PFONT;
     } else if (PShape.class.isAssignableFrom(c)) {
-      return (TYPE_PSHAPE);
+      return TYPE_PSHAPE;
     } else if (GeoXPLib.GeoXPShape.class.isAssignableFrom(c)) {
-      return (TYPE_GEOSHAPE);
+      return TYPE_GEOSHAPE;
     } else if (Set.class.isAssignableFrom(c)) {
-      return (TYPE_SET);
+      return TYPE_SET;
     } else if (BitSet.class.isAssignableFrom(c)) {
-      return (TYPE_BITSET);
+      return TYPE_BITSET;
     } else if (ArrayRealVector.class.isAssignableFrom(c)) {
-      return (TYPE_REALVECTOR);
+      return TYPE_REALVECTOR;
     } else if (RealMatrix.class.isAssignableFrom(c)) {
-      return (TYPE_REALMATRIX);
+      return TYPE_REALMATRIX;
     } else if (AtomicLong.class.isAssignableFrom(c)) {
-      return (TYPE_COUNTER);
+      return TYPE_COUNTER;
     } else if (Typeofable.class.isAssignableFrom(c)) {
       try {
-        return (((Typeofable) c.newInstance()).typeof());
-      } catch(Exception e) {
-        return ("X-" + c.getSimpleName().toUpperCase());
+        return "X-" + ((Typeofable) c.getDeclaredConstructor().newInstance()).typeof();
+      } catch (Exception e) {
+        return defaultType(c);
       }
     } else {
-      return (c.getSimpleName().toUpperCase());
+      return defaultType(c);
+    }
+  }
+
+  private static String defaultType(Class c) {
+    String canonicalName = c.getCanonicalName();
+    if (null == canonicalName) {
+      return "X-(Local/Anonymous Class)";
```

Comment:
It would be better to use neither parens nor spaces.

X-Local-Anonymous-Class?